nevergrad/parametrization/data.py

```diff
@@ -84,39 +84,68 @@ class Data(core.Parameter):
 
     Note
     ----
-    More specific behaviors can be obtained throught the following methods:
-    set_bounds, set_mutation, set_integer_casting
+    - More specific behaviors can be obtained throught the following methods:
+      set_bounds, set_mutation, set_integer_casting
+    - if both lower and upper bounds are provided, sigma will be adapted so that the range spans 6 sigma.
+      Also, if init is not provided, it will be set to the middle value.
     """
 
     def __init__(
         self,
         *,
         init: tp.Optional[tp.ArrayLike] = None,
         shape: tp.Optional[tp.Tuple[int, ...]] = None,
+        lower: tp.BoundValue = None,
+        upper: tp.BoundValue = None,
         mutable_sigma: bool = False,
     ) -> None:
-        sigma = Log(init=1.0, exponent=2.0, mutable_sigma=False) if mutable_sigma else 1.0
         super().__init__()
-        self.parameters = Dict(sigma=sigma, recombination="average", mutation="gaussian")
-        err_msg = 'Exactly one of "init" or "shape" must be provided'
-        self.parameters._ignore_in_repr = dict(sigma="1.0", recombination="average", mutation="gaussian")
+        sigma_val = np.array([1.0])
+        # make sure either shape or
+        if sum(x is None for x in [init, shape]) != 1:
+            raise ValueError('Exactly one of "init" or "shape" must be provided')
         if init is not None:
-            if shape is not None:
-                raise ValueError(err_msg)
-            self._value = np.array(init, dtype=float, copy=False)
-        elif shape is not None:
-            assert isinstance(shape, tuple) and all(
-                isinstance(n, int) for n in shape
-            ), f"Shape incorrect: {shape}."
-            self._value = np.zeros(shape, dtype=float)
+            init = np.array(init, dtype=float, copy=False)
         else:
-            raise ValueError(err_msg)
+            assert isinstance(shape, (list, tuple)) and all(
+                isinstance(n, int) for n in shape
+            ), f"Incorrect shape: {shape}."
+            init = np.zeros(shape, dtype=float)
+            if lower is not None and upper is not None:
+                init += (lower + upper) / 2.0
+        self._value = init
         self.add_layer(_layering.ArrayCasting())
+        # handle bounds
+        num_bounds = sum(x is not None for x in (lower, upper))
+        layer: tp.Any = None
+        if num_bounds:
+            from . import _datalayers
+
+            layer = _datalayers.Bound(
+                lower=lower, upper=upper, uniform_sampling=init is None and num_bounds == 2
+            )
+            if num_bounds == 2:
+                sigma_val = (layer.bounds[1] - layer.bounds[0]) / 6
+        # set parameters
+        if not mutable_sigma:
+            sigma: tp.Any = sigma_val[0] if sigma_val.size == 1 else sigma_val
+        elif sigma_val.size == 1:
+            sigma = Log(init=sigma_val[0], exponent=2.0, mutable_sigma=False)
```

Comment:
@teytaud important question here, should sigma be log-distributed? or linearly-distributed but bounded by (unreachable) 0?
My feeling is that Log seems to evolve too quickly and be unstable, then again, I'm not that sure (eg: TBPSA is using logarithmic updates right?)

Reply:
From all the theory of Anne regarding Markov chains and the experimental works by Rechenberg and others, I would say yes, go for log.

Reply:
from @teytaud coeff to be updated, but still log indeed